iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -51,6 +52,130 @@ inline FixedPositionContainer<T, CAPACITY>::~FixedPositionContainer() noexcept
     }
 }
 
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(const FixedPositionContainer& rhs) noexcept
+{
+    *this = rhs;
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(FixedPositionContainer&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>&
+FixedPositionContainer<T, CAPACITY>::operator=(const FixedPositionContainer& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IndexType i = Index::FIRST;
+        ConstIterator rhs_it = rhs.begin();
+
+        // copy using copy assignment util rhs iter becomes invalid
+        for (; rhs_it.to_index() != Index::INVALID; i++, rhs_it++)
+        {
+            if (m_status[i] == SlotStatus::USED)
+            {
+                // clean original data by it's dtor
+                erase(i);
+            }
+            m_data[i] = *rhs_it;
+            m_status[i] = SlotStatus::USED;
+            // can this line be removed?
+            m_next[i] = static_cast<IndexType>(i + 1U);
+        }
+
+        // correct next
+        m_next[i] = Index::INVALID;
+
+        // erase rest USED element in rhs, m_next and m_status will be also modified by erase()
+        while (i < Index::INVALID)
+        {
+            if (m_status[i] == SlotStatus::USED)
+            {
+                break;
+            }
+
+            i++;
+        }
+
+        // if rest USED slot found
+        if (i != Index::INVALID)
+        {
+            for (Iterator lhs_it = this->iter_from_index(i); lhs_it.to_index() != Index::INVALID; lhs_it++)
+            {
+                erase(lhs_it);
+            }
+        }
```

Comment:
`m_next` must also be updated for the free slots. I guess a simple for loop similar to the one above were the values are copied will work.

Reply:
See e28c2d7